plugins/fileservices/fileservices.cpp

```diff
@@ -523,31 +523,63 @@ FILESERVICES_API void FILESERVICES_CALL fsSetReadOnly(ICodeContext *ctx, const c
     throw error.getClear();
 }
 
-
-FILESERVICES_API void FILESERVICES_CALL fsRenameLogicalFile(ICodeContext *ctx, const char *oldname, const char *newname)
+FILESERVICES_API void FILESERVICES_CALL implementRenameLogicalFile(ICodeContext *ctx, const char *oldname, const char *newname, const bool overwrite)
 {
     StringBuffer lfn, nlfn;
     constructLogicalName(ctx, oldname, lfn);
     constructLogicalName(ctx, newname, nlfn);
 
     IDistributedFileTransaction *transaction = ctx->querySuperFileTransaction();
     Linked<IUserDescriptor> udesc = ctx->queryUserDescriptor();
+
+    IDistributedFileDirectory &distributedDirectory = queryDistributedFileDirectory();
+    if (overwrite)
+    {
+        if ( distributedDirectory.exists(nlfn.str(), ctx->queryUserDescriptor(), false, false) &&
+             distributedDirectory.exists(lfn.str(), ctx->queryUserDescriptor(), false, false))
+        {
+            StringBuffer uname;
+            PrintLog("Deleting NS logical file %s for user %s", nlfn.str(), udesc?udesc->getUserName(uname).str():"");
+            if (distributedDirectory.removeEntry(nlfn.str(), udesc, transaction, INFINITE, true))
+            {
+                StringBuffer s("DeleteLogicalFile ('");
+                s.append(nlfn);
+                if (transaction->active())
+                    s.append("') added to transaction");
+                else
+                    s.append("') done");
+                WUmessage(ctx, SeverityInformation, NULL, s.str());
+                AuditMessage(ctx, "DeleteLogicalFile", nlfn.str());
+            }
```

Comment:
lines 541 to 552 are same as code in fsDeleteLogicalFile.
should be commoned up, i.e. both call common function, or possibly for this just to calll fsDeleteLogicalFile directly.

Reply:
Changed